### PR TITLE
Generic DOM utils parameters

### PR DESCRIPTION
**Pull request by janbiasi**
* `getComponentByDomNode(node) as Button` is now `getComponentByDomNode<Button>(node)`
* the same thing for `getComponentByDomNodeAsync<Button>(node).then(b => ... )`
* and also for `findComponents<Button>(buttonGroup)`
* this change is not breaking as the default generic parameter value is `GondelComponent` so the casting should work as usual.

### CODE REVIEW

File: packages/core/src/GondelDomUtils.ts [hunk 1]
```diff
@@ -58,29 +58,53 @@ export function stopComponents(domContext?: ArrayLikeHtmlElement, namespace: str
   components.forEach(component => component.stop!());
 }
 
+/**
+ * Checks if a component is mounted on a certain DOM node
+ */
+export function componentIsMounted(
```
[the remark this answers is not in the document]
Comment (jantimon) [reply packages/core/src/GondelDomUtils.ts]: Okay I am fine with both 👍

**Comment by jantimon** [comment]
Cool 👍- one last question is it still possible to call `getComponentByDomNode(node)` without setting the generic? and if yes will the result be of type GondelComponent?

**Comment by janbiasi** [comment]
> Is it still possible to call getComponentByDomNode(node) without setting the generic? and if yes will the result be of type GondelComponent?

Yes for sure it is! That's the default behaviour, short example:

```ts
// button is of type 'GondelComponent'
const a = getComponentByDomNode(buttonNode);

// button is of type 'Button'
const b = getComponentByDomNode<Button>(buttonNode);
```

**Sidenote**: I didn't see any documentation regarding the `dom-utils`, I think we should expand the documentation in the next steps for better usability.

**Comment by jantimon** [comment]
Merged 👍

**Pull request closed without merging**